File: Clase 12/PYTHON/ejemplo_gatos/Gato.py
class Gato:
    # Los atributos no son necesarios de declarar

    #metodos
    #Constructor que en python se llama __init__() y requiere del componente self que hace referencia al objeto mismo (todos los metodos tienen que agregar el self para pertenecer a la clase)
    def __init__(self, name:str, sex:str, age:int, weight:int, color:str, texture:str):
         # tipo de dato None, o nulo, hace referencia a que es una variable opcional
        self.name = name
        self.sex = sex
        self.age = age
        self.weight = weight
        self.color = color
        self.texture = texture
    
    def eat(self):
        print(f"{self.name} esta comiendo...")
    
    def move(self):
        print(f"{self.name} se esta moviendo...")
    
    def meow(self):
        print(f"{self.name} hace miauuuuuu...")
    
    def purr(self):
        print(f"{self.name} esta ronrroneando...")

    def hunt_mice(self):
        print(f"{self.name} esta capturando ratoncillos...")

File: Clase 12/JAVA/EjemploGatos/Gato.java
package EjemploGatos;

class Gato{
    //Atributos
    String name;
    String sex;
    int age;
    int weight;
    String color;
    String texture;

    //Metodos
    //El primer metodo siempre es el constructor que en Java se llama de la misma manera que la clase
    Gato(String name, String sex, int age, int  weight, String color, String texture){
        //Un constructor permite crear toda la informacion y metodos  a un objeto.
        //this hace referencia a que son elementos que pertenecen al objeto utilizado
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    void eat(){
        System.out.println(this.name + " esta comiendo");
    }

    void move(){
        System.out.println(this.name + " se esta moviendo...");
    }

    void meow(){
        System.out.println(this.name + " hace miauuuu");
    }

    void purr(){
        System.out.println(this.name + " esta ronrroneando");
    }
    void hunt_mice(){
        System.out.println(this.name + " capturo a un ratoncito...");
    }

}